wasmtime-wasi/src/syscalls.rs

```diff
@@ -323,7 +323,7 @@ syscalls! {
         let mut host_environ = Vec::new();
         host_environ.resize((*argv_environ).environ_count + 1, ptr::null_mut());
 
-        let e = host::wasmtime_ssp_environ_get(argv_environ, host_environ.as_mut_ptr(), host_environ_buf);
+        let e = host_impls::wasmtime_ssp_environ_get(&mut *argv_environ, host_environ.as_mut_slice(), host_environ_buf);
```

Comment:
If we replaced passing `host_environ_buf` here with passing a slice formed from `slice::from_raw_parts(host_environ_buf, environ_buf_len)`, would that allow us to remove the `unsafe` from the new `wasmtime_ssp_environ_get`?

Ditto for `args_get`.

Reply:
OK; there's still two last unsafe parts after that: 1) computing `offset()`, which at worst will cause a panic when computing the subslice for the host_argv pointers, and 2) copying the argv_environ's buf, which has unchecked bounds,  into the host buf